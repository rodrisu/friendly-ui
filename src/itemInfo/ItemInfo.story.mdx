import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks'

import { BASIC } from '../_utils/taxonomy'
import { ClockIcon } from '../icon/clockIcon'
import { ItemInfo } from './index'

<Meta title={`${BASIC}/Items Info`} />

# ItemInfo

<Canvas>
  <Story name="With secondary infos">
    <ItemInfo
      mainTitle="Main title"
      mainInfo="Main info"
      icon={<ClockIcon />}
      tag={<li />}
      aria-label="Custom aria label"
    />
  </Story>
</Canvas>

<Canvas>
  <Story name="Without secondary infos">
    <ItemInfo
      mainTitle="Main title"
      icon={<ClockIcon />}
      tag={<li />}
      aria-label="Custom aria label"
    />
  </Story>
</Canvas>


## Specifications

...

## Usage

```jsx
import { ItemInfo } from '@blablacar/ui-library/build/itemInfo'

<ItemInfo
  mainTitle="Main title"
  mainInfo="Main info"
  icon={<ClockIcon />}
  tag={<li />}
  aria-label="Custom aria label"
/>
```

<ArgsTable of={ItemInfo} />
