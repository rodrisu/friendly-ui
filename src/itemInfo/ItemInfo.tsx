import React from 'react'

import { Item } from '../_internals/item'
import { A11yProps, pickA11yProps } from '../_utils/interfaces'
import { NormalizeProps } from '../layout/layoutNormalizer'

export type ItemInfoProps = A11yProps &
  NormalizeProps &
  Readonly<{
    className?: string
    mainTitle: string
    mainInfo?: string
    icon?: React.ReactNode
    tag?: JSX.Element
  }>

export const ItemInfo = (props: ItemInfoProps) => {
  const { mainInfo, className, mainTitle, icon, tag, hasHorizontalSpacing = false } = props
  const a11yAttrs = pickA11yProps<ItemInfoProps>(props)

  return (
    <Item
      className={className}
      leftTitle={mainTitle}
      leftBody={mainInfo}
      leftAddon={icon}
      tag={tag}
      hasHorizontalSpacing={hasHorizontalSpacing}
      {...a11yAttrs}
    />
  )
}
