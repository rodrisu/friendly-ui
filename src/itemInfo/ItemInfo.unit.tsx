import React from 'react'
import { shallow } from 'enzyme'

import { ItemInfo } from './index'

const defaultProps = {
  mainInfo: 'Main information',
}

it('Should accept a custom `className`', () => {
  const customClassName = 'custom'
  const wrapper = shallow(<ItemInfo {...defaultProps} className={customClassName} />)
  expect(wrapper.hasClass(customClassName)).toBe(true)
})
