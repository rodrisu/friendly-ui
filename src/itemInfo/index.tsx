export { ItemInfo, ItemInfoProps } from './ItemInfo'
