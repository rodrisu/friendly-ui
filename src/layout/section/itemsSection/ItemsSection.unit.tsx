import React from 'react'
import renderer from 'react-test-renderer'

import { ItemInfo } from '../../../itemInfo'
import { ItemsSection } from './index'

describe('ItemsSection', () => {
  it('should render default items section', () => {
    const section = (
      <ItemsSection>
        <ItemInfo mainTitle="First item" mainInfo="First info" />
        <ItemInfo mainTitle="Second item" mainInfo="Second info" />
      </ItemsSection>
    )
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })

  it('should render non-default items section', () => {
    const props = {
      Tag: <span />,
      className: 'contentinfo',
    }
    const section = (
      <ItemsSection {...props}>
        <ItemInfo mainTitle="First item" mainInfo="First info" />
        <ItemInfo mainTitle="Second item" mainInfo="Second info" />
      </ItemsSection>
    )

    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })
})
