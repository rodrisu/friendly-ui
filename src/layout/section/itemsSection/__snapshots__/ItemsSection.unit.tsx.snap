// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsSection should render default items section 1`] = `
.c3 {
  margin: 0;
  font-weight: 400;
}

.c3.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-body,
.c3.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c3.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c3.kirk-text-subheader,
.c3.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-bodyStrong,
.c3.kirk-text-subheaderStrong,
.c3.kirk-text-titleStrong {
  font-weight: 500;
}

.c2 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
  white-space: pre-line;
}

.c2.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c2.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c2.kirk-item--wrappable {
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c2.kirk-item--wrappable .kirk-item-leftWrapper {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c2.kirk-item--wrappable .kirk-item-leftText {
  margin-right: 24px;
}

.c2.kirk-item--wrappable .kirk-item-rightText {
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: left;
}

.c2 .kirk-item-tagToken {
  display: block;
  padding-bottom: 4px;
}

.c2:disabled {
  cursor: default;
}

a.c2 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c2 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c2 {
  outline: none;
}

.c2 .kirk-item-leftWrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c2 .kirk-text {
  display: block;
}

.c2 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c2 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c2 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c2 .kirk-item-leftAddon,
.c2 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c2 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c2 .kirk-item-rightText--withAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-rightText--withAddon svg {
  margin-right: 8px;
}

.c2.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c2 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c1 .kirk-items-section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0 c1"
  role="presentation"
>
  <div
    className="section-content kirk-items-section-content section-content--noHorizontalSpacing"
  >
    <div
      className="kirk-item c2"
    >
      <span
        className="kirk-item-leftWrapper"
      >
        <span
          className="kirk-item-leftText"
        >
          
          <span
            className="kirk-text kirk-text-title c3"
          >
            First item
          </span>
          <span
            className="kirk-text kirk-text-body c3 kirk-item-body"
          >
            First info
          </span>
        </span>
      </span>
    </div>
    <div
      className="kirk-item c2"
    >
      <span
        className="kirk-item-leftWrapper"
      >
        <span
          className="kirk-item-leftText"
        >
          
          <span
            className="kirk-text kirk-text-title c3"
          >
            Second item
          </span>
          <span
            className="kirk-text kirk-text-body c3 kirk-item-body"
          >
            Second info
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`ItemsSection should render non-default items section 1`] = `
.c3 {
  margin: 0;
  font-weight: 400;
}

.c3.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-body,
.c3.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c3.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c3.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c3.kirk-text-subheader,
.c3.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c3.kirk-text-title,
.c3.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c3.kirk-text-bodyStrong,
.c3.kirk-text-subheaderStrong,
.c3.kirk-text-titleStrong {
  font-weight: 500;
}

.c2 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
  white-space: pre-line;
}

.c2.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c2.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c2.kirk-item--wrappable {
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c2.kirk-item--wrappable .kirk-item-leftWrapper {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c2.kirk-item--wrappable .kirk-item-leftText {
  margin-right: 24px;
}

.c2.kirk-item--wrappable .kirk-item-rightText {
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: left;
}

.c2 .kirk-item-tagToken {
  display: block;
  padding-bottom: 4px;
}

.c2:disabled {
  cursor: default;
}

a.c2 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c2 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c2 {
  outline: none;
}

.c2 .kirk-item-leftWrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c2 .kirk-text {
  display: block;
}

.c2 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c2 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c2 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c2 .kirk-item-leftAddon,
.c2 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c2 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c2 .kirk-item-rightText--withAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .kirk-item-rightText--withAddon svg {
  margin-right: 8px;
}

.c2.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c2 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c1 .kirk-items-section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0 c1 contentinfo"
  role="presentation"
>
  <div
    className="section-content kirk-items-section-content section-content--noHorizontalSpacing"
  >
    <div
      className="kirk-item c2"
    >
      <span
        className="kirk-item-leftWrapper"
      >
        <span
          className="kirk-item-leftText"
        >
          
          <span
            className="kirk-text kirk-text-title c3"
          >
            First item
          </span>
          <span
            className="kirk-text kirk-text-body c3 kirk-item-body"
          >
            First info
          </span>
        </span>
      </span>
    </div>
    <div
      className="kirk-item c2"
    >
      <span
        className="kirk-item-leftWrapper"
      >
        <span
          className="kirk-item-leftText"
        >
          
          <span
            className="kirk-text kirk-text-title c3"
          >
            Second item
          </span>
          <span
            className="kirk-text kirk-text-body c3 kirk-item-body"
          >
            Second info
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
